src/pages/upload/index.tsx

```diff
@@ -74,88 +86,206 @@ const fileUploadReducer = (state: any, action: any) => {
+
+  return (
+    <Formik
+      initialValues={initialValues}
+      onSubmit={async (values, actions) => {
+        const response = await axios.post('api/sanity/lessons/create', {
+          lessons: values.lessons,
+        })
+
+        console.log({response})
+      }}
+    >
+      {(props) => <Upload {...props} instructors={instructors} />}
+    </Formik>
+  )
+}
+
+const Upload: React.FC<FormikProps<FormProps> & {instructors: Instructor[]}> = (
+  props,
+) => {
+  const {instructors, ...formikProps} = props
+
+  const {values, setFieldValue} = formikProps
+  const [fileUploadState, dispatch] = React.useReducer(fileUploadReducer, {
+    files: [],
+  })
+  const [lessonMetadata, setLessonMetadata] = React.useState<LessonMetadata[]>(
+    [],
+  )
   const {viewer} = useViewer()
 
   const [courseInstructorId, setCourseInstructorId] = React.useState<string>(
     viewer?.instructor_id || instructors[0]?.eggheadInstructorId,
   )
 
+  React.useEffect(() => {
+    setFieldValue('lessons', lessonMetadata)
+  }, [lessonMetadata, setFieldValue])
+
   return (
-    <div>
-      <ReactS3Uploader
-        ref={uploaderRef}
-        multiple
-        //if we set this to `false` we can list all the files and
-        //call `uploaderRef.current.uploadFile()` when we are ready
-        autoUpload={true}
-        signingUrl={SIGNING_URL}
-        // @ts-ignore
-        signingUrlHeaders={getAuthorizationHeader()}
-        accept="video/*"
-        scrubFilename={(fullFilename) => {
-          //filename with no extension
-          const filename = fullFilename.replace(/\.[^/.]+$/, '')
-          //remove stuff s3 hates
-          const scrubbed = `${filename}-${uuid.generate()}`
-            .replace(/[^\w\d_\-.]+/gi, '')
-            .toLowerCase()
-          //rebuild it as a fresh new thing
-          return `${scrubbed}.${fileExtension(fullFilename)}`
-        }}
-        preprocess={(file, next) => {
-          dispatch({
-            type: 'add',
-            fileUpload: {
-              file,
-              percent: 0,
-              message: 'waiting to upload',
-            },
-          })
-
-          next(file)
-        }}
-        onProgress={(percent, message, file) => {
-          dispatch({type: 'progress', file, percent, message})
-        }}
-        onError={(message) => console.log(message)}
-        onFinish={(signResult, file) => {
-          const fileUrl = signResult.signedUrl.split('?')[0]
-          console.debug(fileUrl, signResult.publicUrl, file)
-        }}
-      />{' '}
-      {state.files.map((file) => {
-        return (
-          <div key={file.name}>
-            {file.file.name} {file.percent} {file.message}
+    <div className="min-h-full flex">
+      <div className="flex-1 flex flex-col justify-center py-12 px-4 sm:px-6 lg:flex-none lg:px-20 xl:px-24">
+        <div>
+          <h2 className="mt-6 text-center text-3xl font-extrabold text-gray-900">
+            egghead Course Builder
+          </h2>
```

Comment:
I dropped in some Tailwind UI read-made styles to make the whole thing nicer to look at and work with. This isn't intended to be the long-term styling, but a reasonable starting point until we want an official design pass.